cypress_test/integration_tests/desktop/writer/top_toolbar_spec.js

```diff
@@ -185,6 +185,79 @@ describe('Top toolbar tests.', function() {
 			.should('have.attr', 'align', 'justify');
 	});
 
+	it('Apply Line spacing: 1', function() {
+		cy.get('#tb_editbar_item_linespacing')
+			.click();
+
+		helper.reselectAllText();
+		
+		cy.get('.menu-text').eq(0)
+			.click();
+
+		helper.reselectAllText();
+
+		cy.get('#copy-paste-container p')
+			.should('have.attr', 'style', 'margin-bottom: 0in; font-style: normal; font-weight: normal; line-height: 100%');
+	});
+
+	it('Apply Line spacing: 1.5', function() {
+		cy.get('#tb_editbar_item_linespacing')
+			.click();
+
+		helper.reselectAllText();
```

Comment:
I feel like you overuse these text selection methods. There is a beforeEach() method in this test suite which is called for all tests once. This beforeEach() contains one text selection, so you don't need to do a new selection before applying the paragraph formatting. Check other tests, how often they use a selection method.